pynest/nest/tests/test_connect_array.py

```diff
@@ -0,0 +1,110 @@
+# -*- coding: utf-8 -*-
+#
+# test_connect_array.py
+#
+# This file is part of NEST.
+#
+# Copyright (C) 2004 The NEST Initiative
+#
+# NEST is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# NEST is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Tests of connection with rules fixed_indegree or fixed_outdegree
+and parameter arrays in syn_spec
+"""
+
+import unittest
+import nest
+import numpy
+
+
+@nest.check_stack
+class ConnectArrayTestCase(unittest.TestCase):
+    """Tests of connections with parameter arrays"""
+
+    def test_Connect_Array(self):
+        """Tests of connections with parameter arrays"""
+        nest.ResetKernel()
+
+        eps = 1.0e-8
+        N = 10
+        K = 3
+        neuron1 = nest.Create('iaf_neuron', N)
+        neuron2 = nest.Create('iaf_neuron', N)
+
+        Warr = [[y*10+x for x in range(K)] for y in range(N)]
+        Darr = [[1.0*y+0.1*x+1.0 for x in range(K)] for y in range(N)]
```

Comment:
You could use integers as well for delays, avoids the need for handling rounding errors.